
The following patch corrects a problem in madplay that can cause it to fail to
decode some frames encoded with LAME (ancillary data contains a LAME tag
marker, but without any tag data).

Robert Leslie <rob@mars.org>
20-Nov-2006


--- player.c.orig	2006-11-20 00:42:12.609647678 -0800
+++ player.c	2006-11-20 00:43:05.741871716 -0800
@@ -1156,8 +1156,10 @@
 	       stream->next_frame - stream->this_frame)
 	player->stats.total_bytes -= frame_size;
 
-      return (player->options & PLAYER_OPTION_SHOWTAGSONLY) ?
-	MAD_FLOW_STOP : MAD_FLOW_IGNORE;
+      if (player->options & PLAYER_OPTION_SHOWTAGSONLY)
+	return MAD_FLOW_STOP;
+      else if (tag->flags)
+	return MAD_FLOW_IGNORE;
     }
     else if (player->options & PLAYER_OPTION_SHOWTAGSONLY)
       return MAD_FLOW_STOP;
